### PR TITLE
Assignment 9 : Error Handling

### PR DESCRIPTION
**Pull request by shafiq005-solin**
# Description
In this assignment, there is a method called divide which will divide the given two integer and returns the quotent value. it also throw Exception when the any value is divided by zero (In other words, Divisor is zero). That error catch in the Interface method. And another method will give the respective elements from the array. It has the possibility to throw Exceptions. And we have to create a custom exception and use that in the respective place. The unhandled exceptions also displayed using a method.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or errors
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/ErrorHandling/Operations.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    public class Operations
+    {
+        /// <summary>
+        /// Act as interface for the user
+        /// </summary>
+        public static void Interface()
+        {
+            int dividend = 0;
+            int divisor = 0;
+            try
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:19]: use a single `try` block and `catch` the corresponding errors separately and then use `finally` block.

```c#
try
{
  //operation
}
catch(InputErrorException ex)
{
  //handle exception
}
catch(DivideByZeroException ex)
{
  //handle exception
}
catch(Exception ex)
{
//  handle exception
}
```

File: src/ErrorHandling/Operations.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                Console.WriteLine("\t\t--------------\nThe default value of an interger is zero!!...\n\t\t---------------");
+            }
+
+            try
+            {
+                int value = Operations.IntegerArray(index);
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:58]:
```suggestion
                int value = IntegerArray(index);
```

File: src/ErrorHandling/Operations.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        {
+            return a / b;
+        }
+
+        /// <summary>
+        /// Returns the value from the array
+        /// </summary>
+        /// <param name="index"> index of the elements</param>
+        /// <returns> value </returns>
+        public static int IntegerArray(int index)
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:83]: Please rename with some valid method name.

File: src/ErrorHandling/Operations.cs [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        {
+            int[] array = new int[5];
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = i * i;
+            }
+
+            return array[index];
+        }
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:92]: Please check the assignment question, go with that flow. (We can discuss regarding this offline)

File: src/ErrorHandling/Operations.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        /// <returns> valid Integer </returns>
+        public static int GetValidInteger(string message)
+        {
+            Console.Write($"{message}");
+            int input = 0;
+            try
+            {
+                input = Convert.ToInt32(Console.ReadLine());
+            }
+            catch (Exception ex)
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:107]: No need of this general exception, please do catch specific parsing exceptions.

File: src/ErrorHandling/Operations.cs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Performs division operation
+        /// </summary>
+        /// <param name="a"> Dividend </param>
+        /// <param name="b"> Divisor </param>
+        /// <returns> Quotient </returns>
+        public static int Divide(int a, int b)
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:73]: This method should return floating number, right?

**Review by MithunSubramani — CHANGES_REQUESTED** [review]

File: src/ErrorHandling/Operations.cs [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        /// Act as interface for the user
+        /// </summary>
+        public static void Interface()
+        {
+            int dividend = 0;
+            int divisor = 0;
+            try
+            {
+                dividend = Operations.GetValidInteger("Enter Divident : ");
+                divisor = Operations.GetValidInteger("Enter Divisor : ");
```
[the remark this answers is not in the document]
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:22]: Operations were removed

File: src/ErrorHandling/Operations.cs (hunk 1 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:19]: single try catch block were implemented

File: src/ErrorHandling/Operations.cs (hunk 2 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:58]: Operations were removed

File: src/ErrorHandling/Operations.cs (hunk 4 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:92]: In assignment tells that to add the changes in the same solution which is used in the task one

File: src/ErrorHandling/Operations.cs (hunk 5 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:107]: General exception were changed

File: src/ErrorHandling/Operations.cs (hunk 6 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:73]: To throw exception, it should return in int. That is the only reason to have that.

File: src/ErrorHandling/Operations.cs (hunk 3 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:83]: Method name was changed to CreateIntegerArray

File: src/ErrorHandling/Operations.cs [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                Console.WriteLine(ex.Message);
+                Console.WriteLine("\t\t--------------\nThe default value of an interger is zero!!...\n\t\t---------------");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            finally
+            {
+                Console.WriteLine("Division Operation is successfully completed!!");
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:41]: I think the message in the finally block is not well suited for the action.

File: src/ErrorHandling/Operations.cs [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        public static int CreateIntegerArray(int index)
+        {
+            int[] array = new int[5];
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = i * i;
+            }
+
+            return array[index];
+        }
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:92]: Try to optimize this logic

File: src/ErrorHandling/readme.md
Comment (MithunSubramani) [inline src/ErrorHandling/readme.md:1]: Update the readme file with the updated method names

**Review by MithunSubramani — CHANGES_REQUESTED** [review]

File: src/ErrorHandling/Operations.cs [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        /// <summary>
+        /// Act as interface for the user
+        /// </summary>
+        public static void OperationsHandler()
+        {
+            int dividend = 0;
+            int divisor = 0;
+            try
+            {
+                dividend = GetValidInteger("Enter Divident : ");
```
Comment (MithunSubramani) [inline src/ErrorHandling/Operations.cs:21]:
```suggestion
                dividend = GetValidInteger("Enter Dividend : ");
```

**Review by MithunSubramani — CHANGES_REQUESTED** [review]

File: src/ErrorHandling/Operations.cs (hunk 8 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:41]: Content in finally were changed

File: src/ErrorHandling/Operations.cs (hunk 9 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:92]: program optimized

File: src/ErrorHandling/Operations.cs (hunk 10 above)
Comment (shafiq005-solin) [reply src/ErrorHandling/Operations.cs:21]: typos were fixed

**Review by MithunSubramani — APPROVED** [review]

**Pull request merged**